freqtrade/commands/hyperopt_commands.py

```diff
@@ -97,18 +111,45 @@ def start_hyperopt_show(args: Dict[str, Any]) -> None:
                                      header_str="Epoch details")
 
 
-def _hyperopt_filter_trials(trials: List, only_best: bool, only_profitable: bool) -> List:
+def _hyperopt_filter_trials(trials: List, filteroptions: dict) -> List:
     """
     Filter our items from the list of hyperopt results
     """
-    if only_best:
+    if filteroptions['only_best']:
         trials = [x for x in trials if x['is_best']]
-    if only_profitable:
+    if filteroptions['only_profitable']:
         trials = [x for x in trials if x['results_metrics']['profit'] > 0]
 
+    if not filteroptions['only_best']:
+        if filteroptions['filter_min_avg_time'] > 0:
+            trials = [x for x in trials if x['results_metrics']['trade_count'] > 0]
```

Comment:
are you filtering out epochs with NaNs here?

Reply:
Yes. It will generate errors trying to compare against NaNs.
Could perhaps be handled differently?

Reply:
no, that's okay, just asking for rationale since I can also be wrong...